src/chart_plugin/ui_helpers/create_arrow.rs

```diff
@@ -22,133 +24,131 @@ pub fn create_arrow(commands: &mut Commands, start: Vec2, end: Vec2, arrow_meta:
+                .build();
+            commands.spawn((
+                ShapeBundle {
+                    path: arrow_path,
+                    ..default()
+                },
+                arrow_meta,
+                Stroke::new(Color::BLACK, 2.0),
+            ));
         }
         ArrowType::ParallelArrow => {
-            eprintln!("Parallel line is not implemented yet")
+            let head_angle = get_angle(arrow_meta.end.pos);
+            let mid_point = parallel_arrow_mid(start, end, arrow_meta);
+            let arrow_path = GeometryBuilder::new()
+                .add(&shapes::Line(start, mid_point.0))
+                .add(&shapes::Line(mid_point.0, mid_point.1))
+                .add(&shapes::Line(mid_point.1, end))
+                .add(&arrow_head(end, head_angle, false))
+                .build();
+            commands.spawn((
+                ShapeBundle {
+                    path: arrow_path,
+                    ..default()
+                },
+                arrow_meta,
+                Stroke::new(Color::BLACK, 2.0),
+            ));
         }
         ArrowType::ParallelDoubleArrow => {
-            eprintln!("Parallel Double Arrow is not implemented yet")
+            let head_angle = get_angle(arrow_meta.end.pos);
+            let tail_angle = get_angle(arrow_meta.start.pos);
+            let mid_point = parallel_arrow_mid(start, end, arrow_meta);
+            let arrow_path = GeometryBuilder::new()
+                .add(&arrow_head(start, tail_angle, true))
+                .add(&shapes::Line(start, mid_point.0))
+                .add(&shapes::Line(mid_point.0, mid_point.1))
+                .add(&shapes::Line(mid_point.1, end))
+                .add(&arrow_head(end, head_angle, false))
+                .build();
+            commands.spawn((
+                ShapeBundle {
+                    path: arrow_path,
+                    ..default()
+                },
+                arrow_meta,
+                Stroke::new(Color::BLACK, 2.0),
+            ));
+        }
+    }
+}
+fn parallel_arrow_mid(start: Vec2, end: Vec2, arrow_meta: ArrowMeta) -> (Vec2, Vec2) {
+    let mid = (start + end) / 2.0;
+    use ArrowConnectPos::*;
+    match (arrow_meta.start.pos, arrow_meta.end.pos) {
+        (Top, Bottom) | (Bottom, Top) => (Vec2::new(start.x, mid.y), Vec2::new(end.x, mid.y)),
+        (Left, Right) | (Right, Left) => (Vec2::new(mid.x, start.y), Vec2::new(mid.x, end.y)),
+        //TODO
+        (Bottom, Left) | (Top, Right) | (Top, Left) | (Bottom, Right) => {
+            (Vec2::new(start.x, end.y), Vec2::new(start.x, end.y))
         }
+        (Left, Bottom) | (Right, Top) | (Left, Top) | (Right, Bottom) => {
+            (Vec2::new(end.x, start.y), Vec2::new(end.x, start.y))
+        }
+        (_, _) => (mid, mid),
+    }
+}
+fn arrow_head(point: Vec2, angle: f32, tail: bool) -> shapes::Polygon {
+    let headlen: f32 = if tail { 10.0 } else { -10.0 };
```

Comment:
I think something wrong with this line... if I change line to `let headlen: f32 = if tail { -10.0 } else { 10.0 };` then bottom and left rectangle connected with arrowheads, but right and top is not.